# generated by patch-package 6.4.8 on 2022-04-03 17:41:55
#
# command:
#   npx patch-package antd-mobile
#
# declared package:
#   antd-mobile: ^5.9.5
#
diff --git a/node_modules/antd-mobile/2x/es/components/toast/methods.js b/node_modules/antd-mobile/2x/es/components/toast/methods.js
index bed0fa1..f540458 100644
--- a/node_modules/antd-mobile/2x/es/components/toast/methods.js
+++ b/node_modules/antd-mobile/2x/es/components/toast/methods.js
@@ -1,18 +1,10 @@
 import React, { createRef, forwardRef, useEffect, useImperativeHandle, useState } from 'react';
 import { resolveContainer } from '../../utils/get-container';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import { InternalToast } from './toast';
 import { mergeProps } from '../../utils/with-default-props';
 const containers = [];
 
-function unmount(container) {
-  const unmountResult = ReactDOM.unmountComponentAtNode(container);
-
-  if (unmountResult && container.parentNode) {
-    container.parentNode.removeChild(container);
-  }
-}
-
 const defaultProps = {
   duration: 2000,
   position: 'center',
@@ -30,7 +22,8 @@ export function show(p) {
   const bodyContainer = resolveContainer(getContainer);
   bodyContainer.appendChild(container);
   clear();
-  containers.push(container);
+  const containerRoot = createRoot(container);
+  containers.push(containerRoot);
   const TempToast = forwardRef((_, ref) => {
     const [visible, setVisible] = useState(true);
     useEffect(() => {
@@ -59,14 +52,15 @@ export function show(p) {
       getContainer: () => container,
       visible: visible,
       afterClose: () => {
-        unmount(container);
+        containerRoot.unmount();
       }
     }));
   });
   const ref = createRef();
-  ReactDOM.render(React.createElement(TempToast, {
+  containerRoot.render(React.createElement(TempToast, {
     ref: ref
-  }), container);
+  }));
+
   return {
     close: () => {
       var _a;
@@ -77,9 +71,9 @@ export function show(p) {
 }
 export function clear() {
   while (true) {
-    const container = containers.pop();
-    if (!container) break;
-    unmount(container);
+    const containerRoot = containers.pop();
+    if (!containerRoot) break;
+    containerRoot.unmount();
   }
 }
 export function config(val) {
diff --git a/node_modules/antd-mobile/2x/es/utils/render-to-body.js b/node_modules/antd-mobile/2x/es/utils/render-to-body.js
index a96ba22..d6bd6af 100644
--- a/node_modules/antd-mobile/2x/es/utils/render-to-body.js
+++ b/node_modules/antd-mobile/2x/es/utils/render-to-body.js
@@ -1,16 +1,14 @@
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 export function renderToBody(element) {
   const container = document.createElement('div');
   document.body.appendChild(container);
 
-  function unmount() {
-    const unmountResult = ReactDOM.unmountComponentAtNode(container);
+  const containerRoot = createRoot(container);
 
-    if (unmountResult && container.parentNode) {
-      container.parentNode.removeChild(container);
-    }
+  function unmount() {
+    containerRoot.unmount();
   }
 
-  ReactDOM.render(element, container);
+  containerRoot.render(element);
   return unmount;
 }
\ No newline at end of file
diff --git a/node_modules/antd-mobile/es/components/toast/methods.js b/node_modules/antd-mobile/es/components/toast/methods.js
index bed0fa1..f540458 100644
--- a/node_modules/antd-mobile/es/components/toast/methods.js
+++ b/node_modules/antd-mobile/es/components/toast/methods.js
@@ -1,18 +1,10 @@
 import React, { createRef, forwardRef, useEffect, useImperativeHandle, useState } from 'react';
 import { resolveContainer } from '../../utils/get-container';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import { InternalToast } from './toast';
 import { mergeProps } from '../../utils/with-default-props';
 const containers = [];
 
-function unmount(container) {
-  const unmountResult = ReactDOM.unmountComponentAtNode(container);
-
-  if (unmountResult && container.parentNode) {
-    container.parentNode.removeChild(container);
-  }
-}
-
 const defaultProps = {
   duration: 2000,
   position: 'center',
@@ -30,7 +22,8 @@ export function show(p) {
   const bodyContainer = resolveContainer(getContainer);
   bodyContainer.appendChild(container);
   clear();
-  containers.push(container);
+  const containerRoot = createRoot(container);
+  containers.push(containerRoot);
   const TempToast = forwardRef((_, ref) => {
     const [visible, setVisible] = useState(true);
     useEffect(() => {
@@ -59,14 +52,15 @@ export function show(p) {
       getContainer: () => container,
       visible: visible,
       afterClose: () => {
-        unmount(container);
+        containerRoot.unmount();
       }
     }));
   });
   const ref = createRef();
-  ReactDOM.render(React.createElement(TempToast, {
+  containerRoot.render(React.createElement(TempToast, {
     ref: ref
-  }), container);
+  }));
+
   return {
     close: () => {
       var _a;
@@ -77,9 +71,9 @@ export function show(p) {
 }
 export function clear() {
   while (true) {
-    const container = containers.pop();
-    if (!container) break;
-    unmount(container);
+    const containerRoot = containers.pop();
+    if (!containerRoot) break;
+    containerRoot.unmount();
   }
 }
 export function config(val) {
diff --git a/node_modules/antd-mobile/es/utils/render-to-body.js b/node_modules/antd-mobile/es/utils/render-to-body.js
index a96ba22..d6bd6af 100644
--- a/node_modules/antd-mobile/es/utils/render-to-body.js
+++ b/node_modules/antd-mobile/es/utils/render-to-body.js
@@ -1,16 +1,14 @@
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 export function renderToBody(element) {
   const container = document.createElement('div');
   document.body.appendChild(container);
 
-  function unmount() {
-    const unmountResult = ReactDOM.unmountComponentAtNode(container);
+  const containerRoot = createRoot(container);
 
-    if (unmountResult && container.parentNode) {
-      container.parentNode.removeChild(container);
-    }
+  function unmount() {
+    containerRoot.unmount();
   }
 
-  ReactDOM.render(element, container);
+  containerRoot.render(element);
   return unmount;
 }
\ No newline at end of file
